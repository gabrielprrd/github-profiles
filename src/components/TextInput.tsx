import React from 'react';

interface Props {
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  placeHolder: string;
}

const TextInput: React.FC<Props> = ({ onChange, placeHolder }) => (
  <input
    type="text"
    onChange={onChange}
    defaultValue=""
    placeholder={placeHolder}
    data-testid="text-input"
    required
  />
);

export default TextInput;
