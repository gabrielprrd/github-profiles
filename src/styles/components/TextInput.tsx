import styled from 'styled-components';

export const TextInput = styled.input`
  border: 4px solid ${({ theme }) => theme.colors.highlightSecondary};
  border-radius: ${({ theme }) => theme.borderRadius};
  padding: ${({ theme }) => theme.padding};
  background: none;
  color: ${({ theme }) => theme.colors.text};
  margin: 10px;
  box-shadow: ${({ theme }) => theme.shadow};
`;
