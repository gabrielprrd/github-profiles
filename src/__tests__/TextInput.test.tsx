import React from 'react';
import ReactDOM from 'react-dom';
import '@testing-library/jest-dom/extend-expect';
import { render, fireEvent, screen, waitFor } from '@testing-library/react';
import renderer from 'react-test-renderer';

// component
import TextInput from '../components/TextInput';

describe('render TextInput', () => {
  it('should render textInput inside a form without crashing', () => {
    const { container } = render(<form />);
    ReactDOM.render(<TextInput />, container.firstChild);
  });

  it('should match snapshot', () => {
    const tree = renderer.create(<TextInput />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
