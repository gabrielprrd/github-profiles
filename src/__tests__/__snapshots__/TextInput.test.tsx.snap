// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render TextInput should match snapshot 1`] = `
<input
  data-testid="text-input"
  defaultValue=""
  required={true}
  type="text"
/>
`;
